### PR TITLE
feat: add AnimatedFAB improvements

### PR DESCRIPTION
**Pull request by Drakeoon**
### Summary

This PR introduces fixes and improvements to original [AnimatedFAB pull request](https://github.com/callstack/react-native-paper/pull/2580)

- [x] Add support for `animateFor: "left"` property
- [x] Add snapshot tests for AnimatedFAB component
- [ ] ~Add integration tests for AnimatedFAB component (new properties)~

### iOS

https://user-images.githubusercontent.com/24693481/134161705-4468b443-a7ae-43b2-8b3e-a5f1eeb26744.mp4

### Android

https://user-images.githubusercontent.com/24693481/134161765-1ce6b348-aa8a-4287-bf0e-0d4c79c1c545.mp4

### CODE REVIEW

**Comment by Drakeoon** [comment]
@lukewalczak I noticed two things:
1. The AnimatedFAB behavior is different depending on the platform. On iOS it expands when we scroll up and collapses when we scroll down. On Android it does the opposite - expands when we scroll down and collapses when we scroll up. Is this an intended behavior?
2. Pressing on the button does not show the ripple effect as on your video. It could be a small bug introduced at some point during development, I'll try to investigate futher

**Comment by lukewalczak** [comment]
> 1. The AnimatedFAB behavior is different depending on the platform. On iOS it expands when we scroll up and collapses when we scroll down. On Android it does the opposite - expands when we scroll down and collapses when we scroll up. Is this an intended behavior?
> 2. Pressing on the button does not show the ripple effect as on your video. It could be a small bug introduced at some point during development, I'll try to investigate futher

1. On both platforms behavior should be the same. According to the documentation: 

_The FAB should not return to an extended FAB until the user scrolls back to the top of the page._

like on the [video](https://kstatic.googleusercontent.com/files/b3bee069a3f5fda3ddcba1a613cb22d6302fe615e441d1de309f44734d01890e5e130fe9a172f6d07052adca0113468e5b1e839be075d1d8a66ee119ec3b6c37) from MD

2. There should be a ripple effect on the whole button.

File: src/components/FAB/AnimatedFAB.tsx [hunk 1]
```diff
@@ -256,8 +256,12 @@ const AnimatedFAB = ({
...
                 width: extendedWidth,
                 opacity: animFAB.interpolate({
-                  inputRange: [distance, 0.9 * distance, 0],
-                  outputRange: [1, 0.15, 0],
+                  inputRange: isAnimatedFromRight
+                    ? [distance, 0.9 * distance, 0]
+                    : [0, 0.9 * distance, distance],
+                  outputRange: isAnimatedFromRight
+                    ? [1, 0.15, 0]
+                    : [0, 0.15, 1],
```
Comment (dsznajder) [inline src/components/FAB/AnimatedFAB.tsx:264]: It looks very common and might be migrated to a helper.

**Comment by Drakeoon** [comment]
@lukewalczak I refactored the AnimatedFAB example, as it was really hard to navigate around the code. I think it's better now, ready for your final review.

I can see there are some conflicts, but I'm not aware of the recent changes on your main branch, so please resolve them and let me know if I can somehow help you with that

**Review by lukewalczak — COMMENTED** [review]
Awesome work 🎉 !

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**